services/QuillLMS/lib/tasks/temporary/translate.rake

```diff
@@ -7,106 +7,318 @@ namespace :translate do
     count = hints.count
     hints.each_with_index do |hint, index|
       puts "translating #{index + 1}/#{count}..."
-      res = Evidence::OpenAI::Translate.run(english_text: hint.description, prompt: hint.prompt)
-      hint.create_translation_mappings
-      hint.english_texts.first.openai_translated_texts.create(
-        translation: res,
-        locale: Translatable::DEFAULT_LOCALE)
+      hint.translate!
     end
 
   end
 
-  task activities: :environment do
-    activities = Activity.where("id in (?)", activity_ids).limit(50)
-    CSV.open("open_ai_spanish_activities.csv", "wb") do |csv|
-      csv << ["english", "spanish"]
-      activities.each_with_index do |activity, index|
-        puts "translating #{index + 1}/50..."
-        html = activity.data["landingPageHtml"]
-        next unless html.present?
+  desc 'translate activities and their questions'
+  task :activities, [:limit] => :environment do |t, args|
+    limit = args[:limit] ? args[:limit].to_i : nil
+    activities = Activity.where("uid in (?)", activity_uids).limit(limit)
```

Comment:
I think that you should just be able to use `where(uid: activity_uids)` and have ActiveRecord automatically figure out the `IN` syntax.